### PR TITLE
remove all type checking from R.concat

### PR DESCRIPTION
**Pull request by davidchambers**
Supersedes #1248

### CODE REVIEW

File: src/concat.js [hunk 1]
```diff
@@ -25,12 +20,4 @@ var _isArray = require('./internal/_isArray');
...
-module.exports = _curry2(function concat(set1, set2) {
-  if (_isArray(set2)) {
-    return _concat(set1, set2);
-  } else if (_hasMethod('concat', set1)) {
-    return set1.concat(set2);
-  } else {
-    throw new TypeError("can't concat " + typeof set1);
-  }
-});
+module.exports = flip(invoker(1, 'concat'));
```
Comment (davidchambers) [inline src/concat.js]: Shall we do this instead?

```javascript
module.exports = _curry2(function concat(a, b) { return a.concat(b); });
```

**Comment by davidchambers** [comment]
((:bell:)) @CrossEye

**Comment by CrossEye** [comment]
Somehow I missed this one when it came through.  Let me think about it a bit.  I'm not completely convinced that we _can't_ do reasonable type checking here.  I just thought that the approach you were suggesting in #1248 was too restrictive.  I'll look at it this evening.

**Comment by CrossEye** [comment]
I guess I'm willing to try this and see if it causes issues.

I'm far from certain about it, though.  I really would rather stop some of the bad cases, but perhaps as we don't do that for any other Semigroups (how would we know if someone's `Maybe List` element could concatenate a `Maybe String`, or that the reverse is much less likely?) it just shouldn't bother me.

So I guess it's :herb: from me.  I'll close #1248, and we can always come back to this if it bites us.

**Pull request merged**